Mathlib/Data/ZMod/Basic.lean

```diff
@@ -491,6 +491,13 @@ noncomputable def ringEquiv [Fintype R] (h : Fintype.card R = n) : ZMod n ≃+*
   RingEquiv.ofBijective _ (ZMod.castHom_bijective R h)
 #align zmod.ring_equiv ZMod.ringEquiv
 
+/-- The unique ring isomorphism between `ZMod p` and a ring `R` cardinality a prime `p`. -/
+noncomputable def ringEquivOfPrime [Fintype R] {p : ℕ} (hp : p.Prime) (hR : Fintype.card R = p) :
+    ZMod p ≃+* R :=
+  have : Nontrivial R := Fintype.one_lt_card_iff_nontrivial.1 (hR ▸ hp.one_lt)
+  have : CharP R p := (CharP.charP_iff_prime_eq_zero hp).2 (hR ▸ Nat.cast_card_eq_zero R)
```

Comment:
Should `p.Prime -> Fintype.card R = p -> CharP R p` be its own lemma?

Reply:
Why not, thanks!

Reply:
I've added it to `Mathlib/Algebra/CharP/CharAndCard.lean`, that is not imported by this file (it seems a better place to have it). Rather then messing up the imports I think we can just keep this proof as it is.